### PR TITLE
feat: add UI support for webhook transform functions

### PR DESCRIPTION
**Pull request by Oluwadaminiola**

### CODE REVIEW

File: web/ui/dashboard/src/app/private/components/create-subscription/create-subscription.component.ts [hunk 1]
```diff
@@ -273,10 +256,20 @@ export class CreateSubscriptionComponent implements OnInit {
 		this.showFilterDialog = true;
 	}
 
+	setupTransformDialog() {
```
Comment (horlah) [inline web/ui/dashboard/src/app/private/components/create-subscription/create-subscription.component.ts:259]: I don't understand this method, why aren't we calling the dialog close/open directly?

**Review by horlah — CHANGES_REQUESTED** [review]

File: web/ui/dashboard/src/app/private/components/create-subscription/create-subscription.component.ts [hunk 2]
```diff
@@ -66,11 +67,13 @@ export class CreateSubscriptionComponent implements OnInit {
...
 	configurations = [
 		{ uid: 'filter_config', name: 'Filter', show: false },
+		{ uid: 'tranform_config', name: 'Transform', show: false },
 		{ uid: 'retry_config', name: 'Retry Logic', show: false }
 	];
 	createdSubscription = false;
 	private rbacService = inject(RbacService);
-    showFilterDialog = false
+	showFilterDialog = false;
+	showTransformDialog = false;
```
[the remark this answers is not in the document]
Comment (Oluwadaminiola) [reply web/ui/dashboard/src/app/private/components/create-subscription/create-subscription.component.ts:76]: The dialog affects the editor, when using the show/close method, the editor was not initialized and was showing up blank hence this route

[the remark this answers is not in the document]
Comment (horlah) [reply web/ui/dashboard/src/app/private/components/create-subscription/create-subscription.component.ts:76]: Hmm, got that

**Review by jirevwe — APPROVED** [review]

File: web/ui/dashboard/src/app/private/components/create-transform-function/create-transform-function.component.ts [hunk 3]
```diff
@@ -20,15 +20,26 @@ export class CreateTransformFunctionComponent implements OnInit {
...
+	payload: any = {
+		id: 'Sample-1',
+		name: 'Sample 1',
+		description: 'This is sample data #1'
+	};
+	setFunction = `// Transform only supports one method, and the function must be named transform as provided below
+// An argument e must be passed into the function which represents the payload to be transformed
+function transform(e) {
+    // Transform function here
+}`;
```
Comment (horlah) [inline web/ui/dashboard/src/app/private/components/create-transform-function/create-transform-function.component.ts:41]: Please update this guy:
`/*  1. While you can write multiple functions, the main function
    called for your transformation is the transform function.
    
    2. The only argument acceptable in the transform function is
    the payload data.
    
    3. The transform method must return a value.
*/

const transform = (payload) => {
    // Transform function here
}
`

**Review by horlah — CHANGES_REQUESTED** [review]

**Comment by horlah** [comment]
@Oluwadaminiola, I'll update the design to accommodate console log messages.

File: web/ui/dashboard/src/app/private/components/create-transform-function/create-transform-function.component.ts (hunk 3 above)
Comment (horlah) [reply web/ui/dashboard/src/app/private/components/create-transform-function/create-transform-function.component.ts:46]: @Oluwadaminiola Did you see this?

**Comment by horlah** [comment]
@Oluwadaminiola The console should be open by default

**Review by horlah — APPROVED** [review]

**Pull request merged**